sys/net/gnrc/network_layer/sixlowpan/nd/gnrc_sixlowpan_nd.c

```diff
@@ -36,7 +36,7 @@ static inline void _rtr_sol_reschedule(gnrc_ipv6_netif_t *iface, uint32_t sec_de
 
 static inline uint32_t _binary_exp_backoff(uint32_t base_sec, unsigned int exp)
 {
-    return genrand_uint32_range(0, (1 << exp) - 1) * base_sec;
+    return genrand_uint32_range(0, (1 << exp)) * base_sec;
```

Comment:
Well, if `iface->rtr_sol_count` equals `GNRC_NDP_MAX_RTR_SOL_NUMOF` exp will be 0, i.e. `(1 << exp) - 1)` would equal 0. Hence, the function will always return 0 for the first call.